packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/config.ts

```diff
@@ -0,0 +1,363 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import * as path from 'path';
+import { Runtime } from './runtime';
+
+/**
+ * Custom resource provider types.
+ */
+export enum ProviderType {
+  /**
+   * `CdkFunction`
+   */
+  CDK_FUNCTION = 'CdkFunction',
+
+  /**
+   * `CdkSingletonFunction`
+   */
+  CDK_SINGLETON_FUNCTION = 'CdkSingletonFunction',
+
+  /**
+   * `CdkCustomResourceProvider`
+   */
+  CDK_CUSTOM_RESOURCE_PROVIDER = 'CdkCustomResourceProvider',
+
+  /**
+   * Do not create a custom resource provider.
+   *
+   * Note: This is used to just move source code for airlifting.
+   */
+  CDK_NO_OP = 'CdkNoOp',
+}
+
+/**
+ * Properites used to generate individual custom resource providers.
+ */
+export interface ProviderProps {
+  /**
+   * The custom resource provider type to generate.
+   */
+  readonly type: ProviderType;
+
+  /**
+   * The source code that will be executed by the provider.
+   */
+  readonly sourceCode: string;
+
+  /**
+   * Runtimes that are compatible with the provider's source code.
+   */
+  readonly compatibleRuntimes: Runtime[];
+
+  /**
+   * The name of the method within your code that the provider calls to execute your function.
+   *
+   * @default 'index.handler'
+   */
+  readonly handler?: string;
+
+  /**
+   * Prevents the source code from being minified and bundled. This is needed for python
+   * files or for source code with a `require` import.
+   *
+   * @default false
+   */
+  readonly preventMinifyAndBundle?: boolean;
+}
+
+export type CustomResourceFrameworkConfig = { [module: string]: { [identifier: string]: ProviderProps[] } };
+
+export const config: CustomResourceFrameworkConfig = {
+  'aws-amplify-alpha': {
+    'asset-deployment-handler': [
+      {
+        type: ProviderType.CDK_NO_OP,
+        sourceCode: path.resolve(__dirname, '..', 'aws-amplify-alpha', 'asset-deployment-handler', 'index.ts'),
+        compatibleRuntimes: [Runtime.NODEJS_18_X],
```

Comment:
Why is this a list? We don't need to do content negotiation, do we? We just stipulate: this Lambda will be using this version. Bam.

Put another way: I see how a list of compatible runtimes makes sense if we say "we support X, Y and Z" and we leave the choice up to the user. But we *don't* leave the choice up to the user!

Extract it to a constant as well.

Reply:
Looks like we use the same function with different handlers. But I agree, we can make `handler` a list instead.